docs/writing_integration_tests.md

```diff
@@ -0,0 +1,97 @@
+# Writing integration tests
+
+## Developing the test source
+
+If you are trying to write a cachi2 integration test, and wish to run cachi2
+against a local source repo, assuming that
+
+- the image "localhost/cachi2:latest" exists and is a valid cachi2 container
+  image
+- your test source is in "~/temp/cachi2-test", **and it is a valid git repo**
+  (i.e. a valid ".git" directory is present)
+
+executing e.g.
+
+```bash
+podman run --rm -ti -v "~/temp/cachi2-test:~/temp/cachi2-test:z" -w "~/temp/cachi2-test" localhost/cachi2:latest
+```
+
+*should* give you a way to properly process the test source with cachi2
+
+## Running pytest locally
+
+Once you have working test sources, you'll need to commit and push them
+somewhere that pytest can clone them from.
+
+We *strongly* recommend making a fork, specific to your test, from one of the
+repos found under the [cachito test repos][] GitHub org (note that there
+are *many* repos there, covering all of the package managers which cachi2 and
+Cachito support) [^1].
+
+Once you have a fork, push to a new branch named after your scenario - now
+pytest will have a proper commit hash in a repo to which you have complete
+access (once your test is complete and passing, you can simply open a PR against
+the repo in the [cachito-testing][] org).
```

Comment:
This is summarized later in the walkthrough in a step-by-step fashion. Can we merge this into the walkthrough so that it doesn't feel like repeating the same information with slightly different wording?